test/rules/avoid_void_async.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_void_async`
+import 'dart:async';
+
+void main() {
+  () {}; // OK
+  () => null; // OK
+  () async {}; // OK
+  () async => null; // OK
+}
+
+void a() async {} // LINT
+void b() {} // OK
+Future<void> c() async {} // OK
+void d() async => null; // LINT
+void e() => null; // OK
+Future<void> f() async => null; // OK
+
+typedef void f1(int x); // OK
+
+typedef Future<void> f2(int x); // OK
+
+class Foo {
+  void a() async {} // LINT
+  void b() {} // OK
+  Future<void> c() async {} // OK
+  void d() async => null; // LINT
+  void e() => null; // OK
+  Future<void> f() async => null; // OK
```

Comment:
I suppose that for completeness sake it might be good to have `static` versions of all of these.

Also, it would be good to add some examples of getters, setters and operators (the latter only on the instance side, of course).